registry/storage/linkedblobstore.go

```diff
@@ -246,13 +246,17 @@ func (lbs *linkedBlobStore) Enumerate(ctx context.Context, ingestor func(digest.
 		}
 
 		// read the digest found in link
-		digest, err := lbs.blobStore.readlink(ctx, filePath)
+		dgst, err := lbs.blobStore.readlink(ctx, filePath)
 		if err != nil {
+			if err == digest.ErrDigestInvalidFormat {
```

Comment:
I am wondering, how did you come across broken digest in your storage? How did it get there?

Reply:
When I run GC, it exits abnormally. To locate the issue, I added some logs and found that the error occurs when parsing the digest while reading some link paths. Then I checked the storage and discovered that these link paths are either empty or corrupt.

Reply:
Ok but that's worrisome right? Because we might be addressing a symptom here, not the root cause

Reply:
yes, but I don't have a good way to reproduce it.

Reply:
If the link paths are empty/corrupt then wouldn't the better idea just to be to remove them and warn rather then leave them be?  I worry we get into a situation with a bunch of these corrupt/empty link files sitting in someones blob store and the only way to remove them would be via manual intervention.